src/esockd_listener.erl

```diff
@@ -117,6 +117,9 @@ handle_cast(Msg, State) ->
     error_logger:error_msg("[~s] Unexpected cast: ~p", [?MODULE, Msg]),
     {noreply, State}.
 
+handle_info({'EXIT', LSock, _}, #state{lsock = LSock} = State) ->
+    error_logger:error_msg("~s Lsocket ~p closed", [?MODULE, LSock]),
+    {stop, lsock_closed, State};
```

Comment:
```suggestion
    {stop, {shutdown, lsock_closed}, State};
```

Reply:
but we need abnormal reason to trigger restart, isn't it?